napari/components/_layer_slicer.py

```diff
@@ -115,9 +119,14 @@ def wait_until_idle(self, timeout: Optional[float] = None) -> None:
             )
 
     def slice_layers_async(
-        self, layers: Iterable[Layer], dims: Dims
-    ) -> Future[dict]:
-        """This should only be called from the main thread.
+        self,
+        layers: Iterable[Layer],
+        dims: Dims,
+        _refresh_sync: bool = False,
+    ) -> Optional[Future[dict]]:
```

Comment:
@kcpevey : I made this method return `None` when there are no async-sliceable layers. Otherwise we actually submit a job with an empty dictionary with seems a little silly. What do you think?

Reply:
The only reason I could see it being worthwhile to go through an empty slicing would be if it fed into an event for `waitUntil` (as mentioned in zulip).

Reply:
> Great, I haven't done a comprehensive read through yet, but I am actively using this branch already.

One thing I haven't added yet is optimized dask caching. I think that adding the cache as part of the request is the only way to do that right now. I was hoping we could hold it in layer slicer itself, but in principle it could be changed (i.e. that should probably happen when data changes, though I don't think it does right now likely because most use cases are read-only).

Reply:
But if no async tasks were submitted, there's nothing to wait for right?

Reply:
I agree with the change. It doesn't make sense to submit empty jobs and if there are no async-sliceable layers, no tasks could have been submitted.